### PR TITLE
docs: add docs 'how to share kcl package' and update docs about 'kpm push'

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

docs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/publish_your_kcl_packages-zh.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+## 步骤 2：创建一个 ghcr.io 令牌
+
+如果您使用 kpm 默认的OCI Registry, 要将 kcl 包推送到 ghcr.io，您需要创建一个用于身份验证的令牌。您可以按照 [ghcr.io 身份验证](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)中的说明进行操作。
+
+## 步骤 3：登录 ghcr.io
+
+在安装了 kpm 并创建了 ghcr.io 令牌后，您需要使用 kpm 登录 ghcr.io。您可以使用以下命令进行操作：
+
+```shell
+kpm login ghcr.io -u <USERNAME> -p <TOKEN> <OCI_REGISTRY>
```
Comment (Peefy) [inline docs/publish_your_kcl_packages-zh.md:20]: Remove dup `<OCI_REGISTRY>`

**Comment by Peefy** [comment]
There are two important questions here:
+ Isn't it necessary to use `ghcr.io` for sharing? I personally think it's best to distinguish between `docker.io`, `ghcr. io`, etc.
+ Format all documents using mdfmt

File: docs/command-reference-zh/1.init.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+$ mkdir my_package 
+
+# 进入目录
+$ cd my_package
+
+# 初始化当前目录为 kcl 包
+$ kpm init
+```
+
+### 初始化一个名为 my_package 的 kcl 包
```
Comment (Peefy) [inline docs/command-reference-zh/1.init.md:40]: Move it to the first example.

File: docs/command-reference-zh/2.add.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+添加一个名为 `k8s` 的依赖，版本为最新版本。
+
+```shell
+kpm add k8s
+```
+
+添加一个名为 `k8s` 的依赖，版本为 `v1.27.2`。
+
+```shell
+kpm add k8s:v1.27.2
```
Comment (Peefy) [inline docs/command-reference-zh/2.add.md:44]: s/kpm add k8s:v1.27.2/kpm add k8s:1.27.2

File: docs/command-reference-zh/5.run.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+```shell
+kpm run /Users/demo/my_package.tar
+```
+
+### 编译一个 kcl 包来自 oci url
+
+`kpm run` 可以用于编译一个 kcl 包从 oci url。
+
+```shell
+kpm run --tag 0.0.1 oci://localhost:5001/test/my_package
```
Comment (Peefy) [inline docs/command-reference-zh/5.run.md:60]: kpm run oci://localhost:5001/test/my_package:0.0.1

File: docs/command-reference-zh/5.run.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+### 编译一个 kcl 包 tar 文件
+
+`kpm run` 可以用于编译一个 kcl 包 tar 文件。
+
+```shell
+kpm run /Users/demo/my_package.tar
+```
+
+### 编译一个 kcl 包来自 oci url
```
Comment (Peefy) [inline docs/command-reference-zh/5.run.md:55]: Remove dup headers.

File: docs/publish_your_kcl_packages.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# How to share your kcl package with others using kpm
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply docs/publish_your_kcl_packages.md]: fixed.

File: docs/publish_your_kcl_packages.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+## Step 2: Create a ghcr.io token
+
+If you are using the default OCI Registry of kpm, to push a kcl package to ghcr.io, you need to create a token for authentication. You can follow the instructions in [ghcr.io authentication](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry).
+
+## Step 3: Log in to ghcr.io
+
+After installing kpm and creating a ghcr.io token, you need to log in to ghcr.io using kpm. You can do this using the following command:
+
+```shell
+kpm login ghcr.io -u <USERNAME> -p <TOKEN> <OCI_REGISTRY>
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply docs/publish_your_kcl_packages.md:20]: fixed.

File: docs/command-reference-zh/1.init.md (hunk 2 above)
Comment (zong-zhe) [reply docs/command-reference-zh/1.init.md]: fixed.

File: docs/command-reference-zh/2.add.md (hunk 3 above)
Comment (zong-zhe) [reply docs/command-reference-zh/2.add.md]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**